docs/specification/draft/server/tools.mdx

```diff
@@ -235,6 +236,69 @@ or data, behind a URI that can be subscribed to or fetched again by the client l
 }
 ```
 
+### Output Schema
+
+The optional `outputSchema` property provides a JSON Schema that describes the expected structure of the `structuredContent` property of a CallToolResult.
+
+When a tool specifies an `outputSchema`:
+
+1. Clients **MUST** validate that the tool result contains a `structuredContent` field whose contents validate against the declared `outputSchema`.
+
+2. Servers **MUST** provide tool results that conform to their declared `outputSchema`s.
+
+Example tool with output schema:
+
+```json
+{
+  "name": "search_database",
+  "description": "Search a database for records",
+  "inputSchema": {
+    "type": "object",
+    "properties": {
+      "query": {
+        "type": "string",
+        "description": "Search query"
+      }
+    },
+    "required": ["query"]
+  },
+  "outputSchema": {
+    "type": "array",
+    "items": {
+      "type": "object",
+      "properties": {
+        "id": { "type": "string" },
+        "title": { "type": "string" },
+        "url": { "type": "string" }
+      },
+      "required": ["id", "title"]
+    }
+  }
+}
+```
+
+Example valid response for this tool:
+
+```json
+{
+  "jsonrpc": "2.0",
+  "id": 5,
+  "result": {
+    "structuredContent": {
```

Comment:
Now that we have a separate key for structuredContent, wouldn't it be preferable for its value to directly conform to the outputSchema rather than having a type and a serialized json? At the very least should we avoid serialization?